agent/consul/state/kvs_test.go

```diff
@@ -1022,6 +1024,163 @@ func TestStateStore_KVSDeleteTree(t *testing.T) {
 	}
 }
 
+func TestStateStore_Watches_PrefixDelete(t *testing.T) {
+	s := testStateStore(t)
+
+	// Create some KVS entries
+	testSetKey(t, s, 1, "foo", "foo")
+	testSetKey(t, s, 2, "foo/bar", "bar")
+	testSetKey(t, s, 3, "foo/bar/zip", "zip")
```

Comment:
I think the more interesting tombstone case would be to add the foo/bar/zip entries last, and then list at foo/bar. It would go like this:

1. Do these inserts:

```
1 foo
2 foo/bar
3 foo/bar/zip
4 foo/bar/zip/zorp
5 foo/bar/zip/zap
6 foo/nope
```

2. List foo/bar, get 5 for the index (from foo/bar/zap)
3. Delete foo/bar/zip by prefix at index 7
4. List foo/bar, get 7 from the tombstone
5. Reap tombstones at 7
6. List foo/bar, index slides back to 2 (from foo/bar)

The current test is kind of skipping the tombstone by looking at a node further down, which is getting the max index from the table.